compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -542,6 +542,35 @@ static void propagateLayoutToContractionOp(
   update(result, changed);
 }
 
+static void propagateLayoutToGatherOp(
+    vector::GatherOp gather,
+    ArrayRef<const DistributionLayout *> operandLattices,
+    ArrayRef<DistributionLayout *> resultLattices,
+    std::function<void(DistributionLayout *, ChangeResult)> update) {
+
+  // We do not support multiple results yet.
+  if (resultLattices.size() != 1)
+    return;
```

Comment:
vector.gather does not have multiple results. You can just assert this or remove this condition.